src/System.Runtime/tests/Performance/Perf.Double.cs

```diff
@@ -12,466 +14,121 @@ namespace System.Tests
-        [InlineData("F9", -2.2250738585072009E-308)]
-        [InlineData("F9", 1.1125369292536E-308)]
-        [InlineData("F9", -0.0)]
-        [InlineData("F9", 0.0)]
-        [InlineData("F15", 104234.343)]
-        [InlineData("F15", double.MinValue / 2)]
-        [InlineData("F15", Math.PI)]
-        [InlineData("F15", Math.E)]
-        [InlineData("F15", double.MaxValue)]
-        [InlineData("F15", double.MinValue)]
-        [InlineData("F15", double.NaN)]
-        [InlineData("F15", double.PositiveInfinity)]
-        [InlineData("F15", double.NegativeInfinity)]
-        [InlineData("F15", 2.2250738585072009E-308)]
-        [InlineData("F15", -2.2250738585072009E-308)]
-        [InlineData("F15", 1.1125369292536E-308)]
-        [InlineData("F15", -0.0)]
-        [InlineData("F15", 0.0)]
-        [InlineData("F17", 104234.343)]
-        [InlineData("F17", double.MinValue / 2)]
-        [InlineData("F17", Math.PI)]
-        [InlineData("F17", Math.E)]
-        [InlineData("F17", double.MaxValue)]
-        [InlineData("F17", double.MinValue)]
-        [InlineData("F17", double.NaN)]
-        [InlineData("F17", double.PositiveInfinity)]
-        [InlineData("F17", double.NegativeInfinity)]
-        [InlineData("F17", 2.2250738585072009E-308)]
-        [InlineData("F17", -2.2250738585072009E-308)]
-        [InlineData("F17", 1.1125369292536E-308)]
-        [InlineData("F17", -0.0)]
-        [InlineData("F17", 0.0)]
-        [InlineData("F50", 104234.343)]
-        [InlineData("F50", double.MinValue / 2)]
-        [InlineData("F50", Math.PI)]
-        [InlineData("F50", Math.E)]
-        [InlineData("F50", double.MaxValue)]
-        [InlineData("F50", double.MinValue)]
-        [InlineData("F50", double.NaN)]
-        [InlineData("F50", double.PositiveInfinity)]
-        [InlineData("F50", double.NegativeInfinity)]
-        [InlineData("F50", 2.2250738585072009E-308)]
-        [InlineData("F50", -2.2250738585072009E-308)]
-        [InlineData("F50", 1.1125369292536E-308)]
-        [InlineData("F50", -0.0)]
-        [InlineData("F50", 0.0)]
-        [InlineData("E", 104234.343)]
-        [InlineData("E", double.MinValue / 2)]
-        [InlineData("E", Math.PI)]
-        [InlineData("E", Math.E)]
-        [InlineData("E", double.MaxValue)]
-        [InlineData("E", double.MinValue)]
-        [InlineData("E", double.NaN)]
-        [InlineData("E", double.PositiveInfinity)]
-        [InlineData("E", double.NegativeInfinity)]
-        [InlineData("E", 2.2250738585072009E-308)]
-        [InlineData("E", -2.2250738585072009E-308)]
-        [InlineData("E", 1.1125369292536E-308)]
-        [InlineData("E", -0.0)]
-        [InlineData("E", 0.0)]
-        public void ToStringWithFormat(string format, double number)
+        public static IEnumerable<object[]> ToStringWithFormat_TestData()
+        {
+            string[] formats =
+            {
+                "R",
+                "G",
+                "G17",
+                "E",
+                "F"
```

Comment:
I would keep `F50` (or maybe `G50`). The internal buffer for the `Number` structure is 50 characters, so this was a test for the maximum number of digits (ensuring that the function doesn't regress across the entire range of digits we can produce).